go/libraries/events/context.go

```diff
@@ -0,0 +1,31 @@
+package events
+
+import (
+	"context"
+	"fmt"
+)
+
+// ContextEventKey key used for storing and retrieving an event from the context.
+const ContextEventKey = "client_metric_event"
```

Comment:
A context event key should be `==` comparable and should not be subject to collisions. So that can just be:

```go
type contextKeyT struct {
}
var ContextEventKey = contextKeyT{}
```

It's also arguable that we shouldn't export `ContextEventKey` here.